### PR TITLE
Copy to csv rework

### PR DESCRIPTION
**Pull request by acquamarin**
This PR improves the existing copy-to-csv:
1. Allow copy to csv to be executed in parallel.
2. Avoid unnecessary flattens on copy to vectors.

E2E performance on copy ldbc-100 to csv file:
60452.38ms (M1 Max, 32GB RAM)

### CODE REVIEW

File: src/include/function/cast/cast_functions.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct CastToString {
     template<typename T>
     static inline std::string castToStringWithVector(
```
Comment (andyfengHKU) [inline src/include/function/cast/cast_functions.h:35]: I would remove `WithVector` from the naming. You can always use same name with overloading.

File: src/processor/operator/physical_operator.cpp [hunk 2]
```diff
@@ -23,8 +23,11 @@ std::string PhysicalOperatorUtils::operatorTypeToString(PhysicalOperatorType ope
     case PhysicalOperatorType::STANDALONE_CALL: {
         return "STANDALONE_CALL";
     }
-    case PhysicalOperatorType::COPY_TO: {
-        return "COPY_TO";
+    case PhysicalOperatorType::COPY_TO_CSV: {
```
Comment (andyfengHKU) [inline src/processor/operator/physical_operator.cpp:26]: We can probably use `LOC` to strip the whole PhysicalTypeToString function from coverage report.

File: src/processor/operator/persistent/copy_to_csv.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        serializer->writeBufferData(std::string(CopyToCSVConstants::DEFAULT_QUOTE));
+        serializer->writeBufferData(strValToWrite);
+        serializer->writeBufferData(std::string(CopyToCSVConstants::DEFAULT_QUOTE));
+    } else {
+        serializer->write(strData, strLen);
+    }
+}
+
+void CopyToCSVLocalState::writeRows() {
+    for (auto i = 0u; i < vectorsToCast.size(); i++) {
```
Comment (andyfengHKU) [inline src/processor/operator/persistent/copy_to_csv.cpp:135]: Cast function should be set at compile time?

File: src/processor/operator/persistent/copy_to_parquet.cpp [hunk 4]
```diff
@@ -3,37 +3,51 @@
 namespace kuzu {
 namespace processor {
 
+using namespace kuzu::common;
+using namespace kuzu::storage;
+
+std::vector<std::unique_ptr<LogicalType>> CopyToParquetInfo::copyTypes() {
```
Comment (andyfengHKU) [inline src/processor/operator/persistent/copy_to_parquet.cpp:9]: We have a similar function is LogicalType::copy

**Review by andyfengHKU — APPROVED** [review]
Take a look at coverage. I feel it should be covered but the report says no.

**Review by ray6080 — APPROVED** [review]

File: src/include/common/constants.h [hunk 5]
```diff
@@ -201,5 +201,15 @@ struct ParquetConstants {
     static constexpr uint64_t MAX_STRING_STATISTICS_SIZE = 10000;
 };
 
+struct CopyToCSVConstants {
```
[the remark this answers is not in the document]
Comment (acquamarin) [reply src/include/common/constants.h:204]: I don't think so, we should separate copyto/copyfrom configs.

**Pull request merged**